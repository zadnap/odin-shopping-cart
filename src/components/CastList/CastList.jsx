import styles from './CastList.module.scss';

function CastList({ casts }) {
  return (
    <section className={styles.castListContainer}>
      <h1 className={styles.title}>The Cast</h1>
      <div className={styles.scroller}>
        <ul className={styles.castList}>
          {casts.map((cast) => (
            <li key={cast.id} className={styles.item}>
              <img
                className={styles.profilePicture}
                src={cast.profilePicture}
                alt={`${cast.name}'s profile picture`}
              />
              <div className={styles.info}>
                <h2 className={styles.actor}>{cast.name}</h2>
                <p className={styles.character}>{cast.character}</p>
              </div>
            </li>
          ))}
        </ul>
      </div>
    </section>
  );
}

export default CastList;
