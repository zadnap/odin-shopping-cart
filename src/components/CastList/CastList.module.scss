@use '@/styles/variables.scss' as *;

.castListContainer {
  .title {
    font-size: 1.8rem;
    margin-bottom: 10px;
  }

  .scroller {
    position: relative;

    &::after {
      content: '';
      position: absolute;
      z-index: 1;
      top: 0;
      right: 0;
      width: 40px;
      height: 100%;
      pointer-events: none;
      background: linear-gradient(to left, #1a1a1a, transparent);
    }

    .castList {
      display: flex;
      gap: var(--gap);
      overflow-y: auto;
      padding-bottom: var(--gap);

      .item {
        background-color: $subbackground-color;
        border-bottom-left-radius: $border-radius;
        border-bottom-right-radius: $border-radius;
        width: 150px;
        flex: 0 0 auto;

        .profilePicture {
          border-top-left-radius: $border-radius;
          border-top-right-radius: $border-radius;
          width: 100%;
          height: 200px;
          object-fit: cover;
        }

        .info {
          padding: 10px;
          .actor {
            font-size: 1.4rem;
          }
          .character {
            font-size: 1.2rem;
          }
        }
      }
    }
  }
}
