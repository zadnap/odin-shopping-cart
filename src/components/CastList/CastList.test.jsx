import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import CastList from './CastList';

const mockCasts = [
  {
    id: 1,
    name: 'Maia Kealoha',
    character: 'Lilo',
    profilePicture:
      'https://media.themoviedb.org/t/p/w138_and_h175_face/jqsKbBF28V2Oq5tKPR5USkNufwC.jpg',
  },
  {
    id: 2,
    name: 'Sydney Agudong',
    character: 'Nani',
    profilePicture:
      'https://media.themoviedb.org/t/p/w138_and_h175_face/3K5hJ3meeClHWsPKetqd9qgyveJ.jpg',
  },
  {
    id: 3,
    name: 'Chris Sanders',
    character: 'Stitch (voice)',
    profilePicture:
      'https://media.themoviedb.org/t/p/w138_and_h175_face/6CtrIOCxggJ5eIAWeFQqd4Hs9FP.jpg',
  },
];

describe('CastList component', () => {
  it('should render to match snapshot', () => {
    const { container } = render(<CastList casts={mockCasts} />);

    expect(container).toMatchSnapshot();
  });
});
