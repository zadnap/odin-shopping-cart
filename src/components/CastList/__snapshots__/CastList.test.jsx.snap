// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CastList component > should render to match snapshot 1`] = `
<div>
  <section
    class="_castListContainer_54003b"
  >
    <h1
      class="_title_54003b"
    >
      The Cast
    </h1>
    <div
      class="_scroller_54003b"
    >
      <ul
        class="_castList_54003b"
      >
        <li
          class="_item_54003b"
        >
          <img
            alt="Maia Kealoha's profile picture"
            class="_profilePicture_54003b"
            src="https://media.themoviedb.org/t/p/w138_and_h175_face/jqsKbBF28V2Oq5tKPR5USkNufwC.jpg"
          />
          <div
            class="_info_54003b"
          >
            <h2
              class="_actor_54003b"
            >
              Maia Kealoha
            </h2>
            <p
              class="_character_54003b"
            >
              Lilo
            </p>
          </div>
        </li>
        <li
          class="_item_54003b"
        >
          <img
            alt="Sydney Agudong's profile picture"
            class="_profilePicture_54003b"
            src="https://media.themoviedb.org/t/p/w138_and_h175_face/3K5hJ3meeClHWsPKetqd9qgyveJ.jpg"
          />
          <div
            class="_info_54003b"
          >
            <h2
              class="_actor_54003b"
            >
              Sydney Agudong
            </h2>
            <p
              class="_character_54003b"
            >
              Nani
            </p>
          </div>
        </li>
        <li
          class="_item_54003b"
        >
          <img
            alt="Chris Sanders's profile picture"
            class="_profilePicture_54003b"
            src="https://media.themoviedb.org/t/p/w138_and_h175_face/6CtrIOCxggJ5eIAWeFQqd4Hs9FP.jpg"
          />
          <div
            class="_info_54003b"
          >
            <h2
              class="_actor_54003b"
            >
              Chris Sanders
            </h2>
            <p
              class="_character_54003b"
            >
              Stitch (voice)
            </p>
          </div>
        </li>
      </ul>
    </div>
  </section>
</div>
`;
